src/wallet/wallet.cpp

```diff
@@ -2480,6 +2480,17 @@ void CWallet::UpdatedTransaction(const uint256 &hashTx)
     }
 }
 
+void CWallet::GetScriptForMining(CScript &script)
+{
+    CReserveKey reservekey(this);
+    reservekey.KeepKey();
```

Comment:
Directly keeping the reserve key for a generate command is okay. It can result in unused but kept reserve keys if no block has been found during `GenerateBitcoins()` (`generate`, `setgenerate`). But this should be okay, at least after we merge HD features (https://github.com/bitcoin/bitcoin/pull/6265).
And this would fix https://github.com/bitcoin/bitcoin/issues/6268